### PR TITLE
Added information about asset caching on CDN

### PR DESCRIPTION
**Pull request by plug-n-play**

### CODE REVIEW

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | plug-n-play/next.js patch-3 | Change |
| - | - | - | - |
| buildDuration | 13.5s | 13.8s | ⚠️ +270ms |
| nodeModulesSize | 65.2 MB | 65.2 MB | ✓ |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | plug-n-play/next.js patch-3 | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.316 | 2.441 | ⚠️ +0.13 |
| / avg req/sec | 1079.57 | 1024.19 | -55.38 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.521 | 1.35 | -0.17 |
| /error-in-render avg req/sec | 1643.42 | 1851.78 | ⚠️ +208.36 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | plug-n-play/next.js patch-3 | Change |
| - | - | - | - |
| main-HASH.js gzip | 6.51 kB | 6.51 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..0538.js gzip | 10.5 kB | 10.5 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 56.9 kB | 56.9 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | plug-n-play/next.js patch-3 | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 5.6 kB | 5.6 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..dule.js gzip | 6.93 kB | 6.93 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 52.4 kB | 52.4 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | plug-n-play/next.js patch-3 | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 26.3 kB | 26.3 kB | ✓ |
| Overall change | 26.3 kB | 26.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | plug-n-play/next.js patch-3 | Change |
| - | - | - | - |
| routerDirect.js gzip | 279 B | 279 B | ✓ |
| withRouter.js gzip | 278 B | 278 B | ✓ |
| _error.js gzip | 3.37 kB | 3.37 kB | ✓ |
| index.js gzip | 222 B | 222 B | ✓ |
| link.js gzip | 2.05 kB | 2.05 kB | ✓ |
| hooks.js gzip | 881 B | 881 B | ✓ |
| _app.js gzip | 1.26 kB | 1.26 kB | ✓ |
| Overall change | 8.34 kB | 8.34 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | vercel/next.js canary  | plug-n-play/next.js patch-3 | Change |
| - | - | - | - |
| index.module.js gzip | 223 B | 223 B | ✓ |
| routerDirect..dule.js gzip | 281 B | 281 B | ✓ |
| withRouter.m..dule.js gzip | 278 B | 278 B | ✓ |
| hooks.module.js gzip | 383 B | 383 B | ✓ |
| _error.module.js gzip | 2.21 kB | 2.21 kB | ✓ |
| link.module.js gzip | 1.52 kB | 1.52 kB | ✓ |
| _app.module.js gzip | 604 B | 604 B | ✓ |
| Overall change | 5.49 kB | 5.49 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | plug-n-play/next.js patch-3 | Change |
| - | - | - | - |
| _buildManifest.js gzip | 270 B | 270 B | ✓ |
| _buildManife..dule.js gzip | 274 B | 274 B | ✓ |
| Overall change | 544 B | 544 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | plug-n-play/next.js patch-3 | Change |
| - | - | - | - |
| index.html gzip | 956 B | 956 B | ✓ |
| link.html gzip | 963 B | 963 B | ✓ |
| withRouter.html gzip | 949 B | 949 B | ✓ |
| Overall change | 2.87 kB | 2.87 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | plug-n-play/next.js patch-3 | Change |
| - | - | - | - |
| buildDuration | 14.7s | 15.2s | ⚠️ +497ms |
| nodeModulesSize | 65.2 MB | 65.2 MB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | plug-n-play/next.js patch-3 | Change |
| - | - | - | - |
| main-HASH.js gzip | 6.51 kB | 6.51 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..0538.js gzip | 10.5 kB | 10.5 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 56.9 kB | 56.9 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | plug-n-play/next.js patch-3 | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 5.6 kB | 5.6 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..dule.js gzip | 6.93 kB | 6.93 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 52.4 kB | 52.4 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | plug-n-play/next.js patch-3 | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 26.3 kB | 26.3 kB | ✓ |
| Overall change | 26.3 kB | 26.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | plug-n-play/next.js patch-3 | Change |
| - | - | - | - |
| routerDirect.js gzip | 279 B | 279 B | ✓ |
| withRouter.js gzip | 278 B | 278 B | ✓ |
| _error.js gzip | 3.37 kB | 3.37 kB | ✓ |
| index.js gzip | 222 B | 222 B | ✓ |
| link.js gzip | 2.05 kB | 2.05 kB | ✓ |
| hooks.js gzip | 881 B | 881 B | ✓ |
| _app.js gzip | 1.26 kB | 1.26 kB | ✓ |
| Overall change | 8.34 kB | 8.34 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | vercel/next.js canary  | plug-n-play/next.js patch-3 | Change |
| - | - | - | - |
| index.module.js gzip | 223 B | 223 B | ✓ |
| routerDirect..dule.js gzip | 281 B | 281 B | ✓ |
| withRouter.m..dule.js gzip | 278 B | 278 B | ✓ |
| hooks.module.js gzip | 383 B | 383 B | ✓ |
| _error.module.js gzip | 2.21 kB | 2.21 kB | ✓ |
| link.module.js gzip | 1.52 kB | 1.52 kB | ✓ |
| _app.module.js gzip | 604 B | 604 B | ✓ |
| Overall change | 5.49 kB | 5.49 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | plug-n-play/next.js patch-3 | Change |
| - | - | - | - |
| _buildManifest.js gzip | 270 B | 270 B | ✓ |
| _buildManife..dule.js gzip | 274 B | 274 B | ✓ |
| Overall change | 544 B | 544 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | vercel/next.js canary  | plug-n-play/next.js patch-3 | Change |
| - | - | - | - |
| _error.js | 875 kB | 875 kB | ✓ |
| 404.html | 4.17 kB | 4.17 kB | ✓ |
| hooks.html | 3.79 kB | 3.79 kB | ✓ |
| index.js | 875 kB | 875 kB | ✓ |
| link.js | 913 kB | 913 kB | ✓ |
| routerDirect.js | 905 kB | 905 kB | ✓ |
| withRouter.js | 905 kB | 905 kB | ✓ |
| Overall change | 4.48 MB | 4.48 MB | ✓ |

</details>

</details>

File: docs/deployment.md [hunk 1]
```diff
@@ -70,6 +70,8 @@ Make sure your `package.json` has the `"build"` and `"start"` scripts:
 
 `next build` builds the production application in the `.next` folder. After building, `next start` starts a Node.js server that supports [hybrid pages](/docs/basic-features/pages.md), serving both statically generated and server-side rendered pages.
 
+All Next.js assets are hashed by default. This means if you use a CDN, with every new deployment, a CDN purge wouldn't be required.
```
Comment (ijjk) [inline docs/deployment.md:73]:
```suggestion
All Next.js assets are hashed by default. This means if you use a CDN that doesn't automatically handle deployment caches like Vercel, with every new deployment, a CDN purge wouldn't be required for files in `/_next` although is still needed for any resources like SSR'd pages outside of the `/_next` folder.
```

We might want to make sure to stress this only applies to `/_next` assets and resources outside of there still might require a purge if not hashed or prevented from being cached

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | plug-n-play/next.js patch-3 | Change |
| - | - | - | - |
| buildDuration | 11.6s | 11.6s | ⚠️ +47ms |
| nodeModulesSize | 65.2 MB | 65.2 MB | ✓ |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | plug-n-play/next.js patch-3 | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 1.988 | 1.942 | -0.05 |
| / avg req/sec | 1257.29 | 1287.58 | ⚠️ +30.29 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.159 | 1.128 | -0.03 |
| /error-in-render avg req/sec | 2157.83 | 2215.52 | ⚠️ +57.69 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | plug-n-play/next.js patch-3 | Change |
| - | - | - | - |
| main-HASH.js gzip | 6.51 kB | 6.51 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..0538.js gzip | 10.5 kB | 10.5 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 56.9 kB | 56.9 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | plug-n-play/next.js patch-3 | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 5.6 kB | 5.6 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..dule.js gzip | 6.93 kB | 6.93 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 52.4 kB | 52.4 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | plug-n-play/next.js patch-3 | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 26.3 kB | 26.3 kB | ✓ |
| Overall change | 26.3 kB | 26.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | plug-n-play/next.js patch-3 | Change |
| - | - | - | - |
| routerDirect.js gzip | 279 B | 279 B | ✓ |
| withRouter.js gzip | 278 B | 278 B | ✓ |
| _error.js gzip | 3.37 kB | 3.37 kB | ✓ |
| index.js gzip | 222 B | 222 B | ✓ |
| link.js gzip | 2.05 kB | 2.05 kB | ✓ |
| hooks.js gzip | 881 B | 881 B | ✓ |
| _app.js gzip | 1.26 kB | 1.26 kB | ✓ |
| Overall change | 8.34 kB | 8.34 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | vercel/next.js canary  | plug-n-play/next.js patch-3 | Change |
| - | - | - | - |
| index.module.js gzip | 223 B | 223 B | ✓ |
| routerDirect..dule.js gzip | 281 B | 281 B | ✓ |
| withRouter.m..dule.js gzip | 278 B | 278 B | ✓ |
| hooks.module.js gzip | 383 B | 383 B | ✓ |
| _error.module.js gzip | 2.21 kB | 2.21 kB | ✓ |
| link.module.js gzip | 1.52 kB | 1.52 kB | ✓ |
| _app.module.js gzip | 604 B | 604 B | ✓ |
| Overall change | 5.49 kB | 5.49 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | plug-n-play/next.js patch-3 | Change |
| - | - | - | - |
| _buildManifest.js gzip | 270 B | 270 B | ✓ |
| _buildManife..dule.js gzip | 274 B | 274 B | ✓ |
| Overall change | 544 B | 544 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | plug-n-play/next.js patch-3 | Change |
| - | - | - | - |
| index.html gzip | 956 B | 956 B | ✓ |
| link.html gzip | 963 B | 963 B | ✓ |
| withRouter.html gzip | 949 B | 949 B | ✓ |
| Overall change | 2.87 kB | 2.87 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | plug-n-play/next.js patch-3 | Change |
| - | - | - | - |
| buildDuration | 14.6s | 14.2s | -379ms |
| nodeModulesSize | 65.2 MB | 65.2 MB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | plug-n-play/next.js patch-3 | Change |
| - | - | - | - |
| main-HASH.js gzip | 6.51 kB | 6.51 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..0538.js gzip | 10.5 kB | 10.5 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 56.9 kB | 56.9 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | plug-n-play/next.js patch-3 | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 5.6 kB | 5.6 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..dule.js gzip | 6.93 kB | 6.93 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 52.4 kB | 52.4 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | plug-n-play/next.js patch-3 | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 26.3 kB | 26.3 kB | ✓ |
| Overall change | 26.3 kB | 26.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | plug-n-play/next.js patch-3 | Change |
| - | - | - | - |
| routerDirect.js gzip | 279 B | 279 B | ✓ |
| withRouter.js gzip | 278 B | 278 B | ✓ |
| _error.js gzip | 3.37 kB | 3.37 kB | ✓ |
| index.js gzip | 222 B | 222 B | ✓ |
| link.js gzip | 2.05 kB | 2.05 kB | ✓ |
| hooks.js gzip | 881 B | 881 B | ✓ |
| _app.js gzip | 1.26 kB | 1.26 kB | ✓ |
| Overall change | 8.34 kB | 8.34 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | vercel/next.js canary  | plug-n-play/next.js patch-3 | Change |
| - | - | - | - |
| index.module.js gzip | 223 B | 223 B | ✓ |
| routerDirect..dule.js gzip | 281 B | 281 B | ✓ |
| withRouter.m..dule.js gzip | 278 B | 278 B | ✓ |
| hooks.module.js gzip | 383 B | 383 B | ✓ |
| _error.module.js gzip | 2.21 kB | 2.21 kB | ✓ |
| link.module.js gzip | 1.52 kB | 1.52 kB | ✓ |
| _app.module.js gzip | 604 B | 604 B | ✓ |
| Overall change | 5.49 kB | 5.49 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | plug-n-play/next.js patch-3 | Change |
| - | - | - | - |
| _buildManifest.js gzip | 270 B | 270 B | ✓ |
| _buildManife..dule.js gzip | 274 B | 274 B | ✓ |
| Overall change | 544 B | 544 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | vercel/next.js canary  | plug-n-play/next.js patch-3 | Change |
| - | - | - | - |
| _error.js | 875 kB | 875 kB | ✓ |
| 404.html | 4.17 kB | 4.17 kB | ✓ |
| hooks.html | 3.79 kB | 3.79 kB | ✓ |
| index.js | 875 kB | 875 kB | ✓ |
| link.js | 913 kB | 913 kB | ✓ |
| routerDirect.js | 905 kB | 905 kB | ✓ |
| withRouter.js | 905 kB | 905 kB | ✓ |
| Overall change | 4.48 MB | 4.48 MB | ✓ |

</details>

</details>

**Review by timneutkens — CHANGES_REQUESTED** [review]
This statement is confusing as it talks about hashes whereas not all files are hashed, some use the buildId approach to make them unique. It's generally best to do a CDN purge after a new deployment as otherwise people that load the page right before a new deployment don't get automatic reloading when navigating to the new page.

**Comment by plug-n-play** [comment]
Intention was to help out anyone who relies on hashes instead of doing a CDN purge with every new build for non-Vercel deployment. Should we modify the statement to make it more clear or remove it altogether and do not write anything about this?

**Comment by timneutkens** [comment]
In general it's common practice to flush the CDN cache once a new deployment is created so I don't think we should include this line as it would confuse users.

**Pull request closed without merging**